From fbf5ab93af9ceb2f4d31e7f4fe8b523b939fc846 Mon Sep 17 00:00:00 2001
From: Ulrik Strid <ulrik.strid@outlook.com>
Date: Tue, 4 Jan 2022 09:48:48 +0100
Subject: [PATCH] pythonPackages.python-miio: 0.5.8 -> 0.5.9.2

The miio integration in home-assistant is broken with the current version. These changes works as a override in my setup.
---
 .../python-modules/python-miio/default.nix           | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/pkgs/development/python-modules/python-miio/default.nix b/pkgs/development/python-modules/python-miio/default.nix
index 46900dd80e3d4..c4af73d3dc62c 100644
--- a/pkgs/development/python-modules/python-miio/default.nix
+++ b/pkgs/development/python-modules/python-miio/default.nix
@@ -24,21 +24,21 @@
 
 buildPythonPackage rec {
   pname = "python-miio";
-  version = "0.5.8";
+  version = "0.5.9.2";
   disabled = pythonOlder "3.6";
   format = "pyproject";
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "sha256-16XEah5rgem/L8A/zo1zPrifrU15VMk652rFLZcvjig=";
+    sha256 = "sha256-AFwarRhFknfwTSvSDGoWE+/mv1KUD2XnWK/xCBqrN4o=";
   };
 
   postPatch = ''
     substituteInPlace pyproject.toml \
-      --replace 'click = "^7"' 'click = "*"' \
-      --replace 'croniter = "^0"' 'croniter = "*"' \
-      --replace 'cryptography = "^3"' 'cryptography = "*"' \
-      --replace 'defusedxml = "^0.6"' 'defusedxml = "*"'
+      --replace 'click = ">=7"' 'click = "*"' \
+      --replace 'croniter = ">=1"' 'croniter = "*"' \
+      --replace 'cryptography = ">=35"' 'cryptography = "*"' \
+      --replace 'defusedxml = "^0"' 'defusedxml = "*"'
   '';
 
   nativeBuildInputs = [
